mysql/infoschema.go

```diff
@@ -375,6 +380,47 @@ func getForeignKeys(conv *internal.Conv, db *sql.DB, table schemaAndName) (forei
 	return foreignKeys, nil
 }
 
+// getIndexes return list all the indexes.
+func getIndexes(conv *internal.Conv, db *sql.DB, table schemaAndName) (indexes []schema.Index, err error) {
+	q := `SELECT DISTINCT INDEX_NAME,COLUMN_NAME,SEQ_IN_INDEX,COLLATION,NON_UNIQUE
+		FROM INFORMATION_SCHEMA.STATISTICS 
+		WHERE TABLE_SCHEMA = ?
+			AND TABLE_NAME = ?
+			AND INDEX_NAME != 'PRIMARY' 
+		ORDER BY INDEX_NAME, SEQ_IN_INDEX;`
+	rows, err := db.Query(q, table.schema, table.name)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var name, col, seq, collation, nonUnique string
+	m := make(map[string][]schema.Key)
+	unique := make(map[string]bool)
+	var keyNames []string
+	for rows.Next() {
+		err := rows.Scan(&name, &col, &seq, &collation, &nonUnique)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
+			continue
+		}
```

Comment:
It's more typical to write this in Go as

if err := rows.Scan(&name, &col, &seq, &collation, &nonUnique); err != nil {
   conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
   continue
}

similarly elsewhere

Reply:
yeah its much cleaner, thanks